### ISSUE TITLE
[CRASH] Deleting an audio file currently opened in Footage Viewer from Project crashes Olive

### ISSUE THREAD

**Issue by FirasH** [olive-editor/olive#1258]
**Commit Hash** 
0.2.0 ac4be090

**Platform**
Windows 10

**Summary**
Deleting an audio file currently opened in Footage Viewer from Project crashes Olive.
This issue does not happen with video files.

**Steps to Reproduce**
1. Go to: `File > Import...` and import an audio file (tested with `.mp3` and `.wav`)
2. Double-click to open the imported audio file in the `Footage Viewer`.
3. Delete the imported audio file from the `Project` list of imported files.

Olive crashes consistently.

<details><summary><strong>Backtrace</strong></summary><pre><code>

</code></pre></details>

**Additional Information**

**Comment by ThomasWilshaw** [comment]
I can confirm this, attached is my backtrace:

<details>
  <summary>Backtrace</summary>

```
olive-editor.exe!olive::AudioPlaybackCache::GetParameters() Line 38
	at C:\Users\Tom\Documents\repos\olive\app\render\audioplaybackcache.h(38)
olive-editor.exe!olive::AudioWaveformView::paintEvent(QPaintEvent * event) Line 66
	at C:\Users\Tom\Documents\repos\olive\app\widget\viewer\audiowaveformview.cpp(66)
Qt5Widgetsd.dll!QWidget::event(QEvent * event) Line 8815
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qwidget.cpp(8815)
Qt5Widgetsd.dll!QApplicationPrivate::notify_helper(QObject * receiver, QEvent * e) Line 3685
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qapplication.cpp(3685)
Qt5Widgetsd.dll!QApplication::notify(QObject * receiver, QEvent * e) Line 3635
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qapplication.cpp(3635)
Qt5Cored.dll!QCoreApplication::notifyInternal2(QObject * receiver, QEvent * event) Line 1075
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qcoreapplication.cpp(1075)
Qt5Cored.dll!QCoreApplication::sendSpontaneousEvent(QObject * receiver, QEvent * event) Line 1483
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qcoreapplication.cpp(1483)
Qt5Widgetsd.dll!QWidgetPrivate::sendPaintEvent(const QRegion & toBePainted) Line 5477
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qwidget.cpp(5477)
Qt5Widgetsd.dll!QWidgetPrivate::drawWidget(QPaintDevice * pdev, const QRegion & rgn, const QPoint & offset, QFlags<enum QWidgetPrivate::DrawWidgetFlag> flags, QPainter * sharedPainter, QWidgetRepaintManager * repaintManager) Line 5427
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qwidget.cpp(5427)
Qt5Widgetsd.dll!QWidgetRepaintManager::paintAndFlush() Line 1012
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qwidgetrepaintmanager.cpp(1012)
Qt5Widgetsd.dll!QWidgetRepaintManager::sync() Line 777
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qwidgetrepaintmanager.cpp(777)
Qt5Widgetsd.dll!QWidgetPrivate::syncBackingStore() Line 1767
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qwidget.cpp(1767)
Qt5Widgetsd.dll!QWidget::event(QEvent * event) Line 8979
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qwidget.cpp(8979)
Qt5Widgetsd.dll!QMainWindow::event(QEvent * event) Line 1342
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\widgets\qmainwindow.cpp(1342)
Qt5Widgetsd.dll!QApplicationPrivate::notify_helper(QObject * receiver, QEvent * e) Line 3685
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qapplication.cpp(3685)
Qt5Widgetsd.dll!QApplication::notify(QObject * receiver, QEvent * e) Line 3635
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\widgets\kernel\qapplication.cpp(3635)
Qt5Cored.dll!QCoreApplication::notifyInternal2(QObject * receiver, QEvent * event) Line 1075
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qcoreapplication.cpp(1075)
Qt5Cored.dll!QCoreApplication::sendEvent(QObject * receiver, QEvent * event) Line 1471
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qcoreapplication.cpp(1471)
Qt5Cored.dll!QCoreApplicationPrivate::sendPostedEvents(QObject * receiver, int event_type, QThreadData * data) Line 1815
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qcoreapplication.cpp(1815)
Qt5Cored.dll!QEventDispatcherWin32::sendPostedEvents() Line 1052
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qeventdispatcher_win.cpp(1052)
qwindowsd.dll!000007feaa74ad04()
Qt5Cored.dll!QEventDispatcherWin32::processEvents(QFlags<enum QEventLoop::ProcessEventsFlag> flags) Line 525
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qeventdispatcher_win.cpp(525)
qwindowsd.dll!000007feaa74acc4()
Qt5Cored.dll!QEventLoop::processEvents(QFlags<enum QEventLoop::ProcessEventsFlag> flags) Line 139
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qeventloop.cpp(139)
Qt5Cored.dll!QEventLoop::exec(QFlags<enum QEventLoop::ProcessEventsFlag> flags) Line 225
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qeventloop.cpp(225)
Qt5Cored.dll!QCoreApplication::exec() Line 1383
	at C:\Users\Tom\Documents\repos\vcpkg\buildtrees\qt5-base\src\5.14.2-8b6659c0d6\src\corelib\kernel\qcoreapplication.cpp(1383)
olive-editor.exe!main(int argc, char * * argv) Line 161
	at C:\Users\Tom\Documents\repos\olive\app\main.cpp(161)
olive-editor.exe!invoke_main() Line 79
	at D:\agent\_work\9\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl(79)
olive-editor.exe!__scrt_common_main_seh() Line 288
	at D:\agent\_work\9\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl(288)
olive-editor.exe!__scrt_common_main() Line 331
	at D:\agent\_work\9\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl(331)
olive-editor.exe!mainCRTStartup() Line 17
	at D:\agent\_work\9\s\src\vctools\crt\vcstartup\src\startup\exe_main.cpp(17)
kernel32.dll!BaseThreadInitThunk()
ntdll.dll!RtlUserThreadStart()

```
</details>

And error message:

`Exception thrown at 0x000000013F62A17F in olive-editor.exe: 0xC0000005: Access violation reading location 0x0000000000000038.`

**Comment by itsmattkc** [comment]
Is the problem fixed in 3b6ebb6e300468b2c4d75231aeb0bb5235171419?

**Comment by FirasH** [comment]
> Is the problem fixed in [3b6ebb6](https://github.com/olive-editor/olive/commit/3b6ebb6e300468b2c4d75231aeb0bb5235171419)?

Yes, fixed.
Thanks :+1:
